src/HCSR04.cpp

```diff
@@ -19,22 +18,25 @@ int HCSR04::getDistance() {
     hwlib::wait_ms(10);
     trigger_pin.set(0);
 
+    // Get current time
+    long long int duration = hwlib::now_us();
+
     // Receive pulse
     bool received = echo_pin.get();
 
-    // Get current time
-    int duration = (int)hwlib::now_us();
-
-    while (!received) {
-        received = echo_pin.get();
-    }
-    while (received) {
-        received = echo_pin.get();
+    // If the first pulse hasn't been received yet, wait until first pulse is found.
+    if (!received) {
```

Comment:
This is an unnecessary extra check. Just check if the signal goes high directly.